server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -944,4 +944,11 @@ public void test_all_eq_on_nested_array_ref_with_automatic_dimension_leveling()
                 );
         }
     }
+
+    @Test
+    public void test_cannot_query_index_ref_with_analyzer_that_is_not_used_to_index_with() {
+        assertThatThrownBy(() -> convert("match(name, 'foo') USING phrase WITH (analyzer='dummy')"))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Column 'name' was indexed with keyword analyzer, searching with a different analyzer, 'dummy' is not supported");
+    }
```

Comment:
Please add another test that enters the `if` part of the check: https://github.com/crate/crate/pull/17703/files#diff-3f5f1c1f75e4403d6217cb0ad10193a43aac36e455a185463dc3b7475658cfdcR129 (not keyword analyzer)